extract.py

```diff
@@ -215,9 +216,9 @@ def gather_movie_data(self):
 
         omdb_data = self._thread_pool(self._omdb_fetch_data, {'max_range': math.ceil(len(self._new_movies) / MAX_WORKERS), 'max_workers': MAX_WORKERS, 'type': 1})
         
-        # kafka_interface.produce_to_topic('nosaqtgg-omdb-api', omdb_data)
+        kafka_interface.produce_to_topic('nosaqtgg-omdb-api', omdb_data)
         self._update_json_file('omdb_data.json', omdb_data)
 
 # Run section
 extractor = Extract(5)
-extractor.start()
+extractor.start()
```

Comment:
the thread pool should be a class.
also - you should make sure you free all threads and proccesses you are using so you wont have memory leaks